### PR TITLE
Improvements #1

### PR DESCRIPTION
**Pull request by FKasy**
- create reusable React Components
- structure application in clear modules
- use modern React features
- make use of PropTypes for type-checking
- clean up all eslint, prettier an webpack config
- make eslint warn instead of error
- move all babel and webpack packages into devDependencies
- import bulma directly in main scss file
- create inital scss modules
- add loading spinner to button
- animate cards only when search was succesfull

### CODE REVIEW

**Comment by PhilippMatthes** [comment]
Please use a better structured commit message, i.e. find a meaningful one lined tl;dr and list all detailled changes below. If it is hard for you to summarize all changes in a one liner, this might be a hint that you need to split your commit into few grouped commits, each named meaningfully. Please don‘t use „:“s as if you are opening a list. One should be able to identify the purpose of a commit just by viewing the shortened version, and if necessary get more detailled information on demand.

File: src/helper/DateHelper.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { formatRelative } from "date-fns";
+
+export default function stringifyTimestamp(timestamp) {
+  return typeof timestamp === "undefined" ? "n/a" : formatRelative(new Date(timestamp), Date.now());
```
Comment (PhilippMatthes) [inline src/helper/DateHelper.js]: Use explicit if clauses to increase readability and to reduce line length and complexity.

File: src/styles/_form.scss [hunk 2]
```diff
@@ -2,45 +2,57 @@
...
-  height: 50px;
-  font-size: 80%;
-  transition: all 0.2s;
+  .button {
+    border-radius: 5rem;
+    letter-spacing: 0.1rem;
+    font-weight: bold;
+    height: 50px;
+    font-size: 80%;
+    transition: all 0.2s;
```
Comment (PhilippMatthes) [inline src/styles/_form.scss]: This appears to be technical debt, can you restrict the animation on particular properties?

File: package.json [hunk 3]
```diff
@@ -24,29 +24,29 @@
   },
   "homepage": "https://jsexam.github.io/jsExam/",
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (FKasy) [reply package.json]: The dependencies did **not** change. They were only moved from "dependencies" into "devDependencies". These packages do not contain any malicious dependencies.

File: .stylelintrc [hunk 4]
```diff
@@ -1,7 +1 @@
-{
-  "processors": ["stylelint-processor-styled-components"],
-  "extends": [
-    "stylelint-config-recommended",
-    "stylelint-config-styled-components"
-  ]
-}
\ No newline at end of file
+{}
```
[the remark this answers is not in the document]
Comment (FKasy) [reply .stylelintrc]: Fixed in 4caed4bb870c3e3725f0cbe45278272646862f69

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (FKasy) [reply package.json]: see 561dbca8ff151e97627eddcb9a7a6bead9f36735

File: src/App.js [hunk 5]
```diff
@@ -1,138 +1,78 @@
 import React, { Component } from "react";
-import "./styles/Bulma.sass";
-import "./styles/App.scss";
 import anime from "animejs/lib/anime.es.js"; // Only include Anime.JS in the components you need it.
-import { stringifyTimestamp } from "./helper/Date";
```
[the remark this answers is not in the document]
Comment (FKasy) [reply src/App.js]: Fixed in a0fd16f9fc49521c0a50f3e843dfd8d4614efdd4

File: src/helper/DateHelper.js (hunk 1 above)
Comment (FKasy) [reply src/helper/DateHelper.js]: Fixed in a0fd16f9fc49521c0a50f3e843dfd8d4614efdd4

File: src/styles/_form.scss (hunk 2 above)
Comment (FKasy) [reply src/styles/_form.scss]: Removed in 8a5bd8b8a622fbaaaa160c8ef53b89dc7a587da0

**Pull request merged**